src/lib/cache/memory.ts

```diff
@@ -66,13 +66,21 @@ export const memoryCache: CacheBackend = {
  * With next-on-pages, the code seems to be isolated between the middleware and the handler.
  * To share the cache between the two, we use a global variable.
  */
-function getMemoryCache(): Map<string, CacheEntry> {
-    // @ts-ignore
-    if (!globalThis.gitbookMemoryCache) {
-        // @ts-ignore
-        globalThis.gitbookMemoryCache = new Map();
+async function getMemoryCache(): Promise<Map<string, CacheEntry>> {
+    let globalThisForMemoryCache: any = globalThis;
+
+    if (process.env.NODE_ENV !== 'test') {
+        // We lazy-load the next-on-pages package to avoid errors when running tests because of 'server-only'.
+        const { getOptionalRequestContext } = await import('@cloudflare/next-on-pages');
+        const cloudflare = getOptionalRequestContext();
+        if (cloudflare) {
+            globalThisForMemoryCache = cloudflare.ctx;
+        }
+    }
```

Comment:
Can we use https://github.com/GitbookIO/gitbook/blob/main/src/lib/waitUntil.ts#L7?

or at least keep things centralize on this file

Reply:
Ah right, good call 👍